docs/marionette.events.md

```diff
@@ -0,0 +1,416 @@
+```
+
+As in [Backbone](http://backbonejs.org/#Events), `listenTo` will pass the object
+it is called on in as the context variable. These behave exactly as in Backbone,
+so using `object.on` will require you to unhook any event handlers yourself to
+prevent memory leaks. Marionette, however, does provide extra helpers as part of
+the view lifecycle that bind and unbind event handlers for you. this is the
+core of Magic Method Binding.
+
+#### Magic Method Binding
+
+The major difference between `Backbone.trigger` and `View.triggerMethod` is
+that `triggerMethod` can fire specially named events on the attached view. For
+instance, a view that has been rendered will fire `view.triggerMethod('render')`
+and call `onRender` - providing a handy way to add behavior to your views.
+
+Determining what method an event will call is easy, we will outline this with an
+example using `before:dom:refresh` though this also works with any custom events
+you want to fire:
+
+1. Split the words around the `:` characters - so `before`, `dom`, `refresh`
+2. Capitalize the first letter of each word - `Before`, `Dom`, `Refresh`
+3. Add a leading `on` - `on`, `Before`, `Dom`, `Refresh`
+4. Mash it into a single call - `onBeforeDomRefresh`
+
+Using this process, `before:dom:refresh` will call the `onBeforeDomRefresh`
+method. Let's see it in action with a custom event:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({
+  onMyEvent: function(myVal) {
+    console.log(myVal);
+  }
+});
+
+var myView = new MyView();
+
+myView.triggerMethod('my:event', 'someValue'); // Logs 'someValue'
+```
+
+As before, all arguments passed into `triggerMethod` will make their way into
+the event handler. Using this method ensures there will be no unexpected
+memory leaks.
+
+### View events and triggers
+
+Views can automatically bind DOM events to methods and View events with `events`
+and `triggers` respectively:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var MyView = Mn.View.extend({
+  events: {
+    'click a': 'showModal'
+  },
+
+  triggers: {
+    'keyup input': 'data:entered'
+  },
+
+  showModal: function(event) {
+    console.log('Show the modal');
+  },
+
+  onDataEntered: function() {
+    console.log('Data was entered');
+  }
+});
+```
+
+For more information, see the [view documentation](./marionette.view.md#binding-to-user-input).
+
+## Child View Events
+
+The [`View`](marionette.view.md) and [`CollectionView`](marionette.collectionview.md)
+are able to monitor and act on events on any children they own. Any events fired
+on a view are automatically propagated to their direct parents as well. Let's
+see a quick example:
+
+```javascript
+var Mn = require('backbone.marionette');
+
+var Item = Mn.View.extend({
+  tagName: 'li',
+
+  triggers: {
+    'click a': 'select:item'
+  }
+});
+
+var Collection = Mn.CollectionView.extend({
+  tagName: 'ul',
+
+  childViewEvents: {
+    'select:item': 'itemSelected'
```

Comment:
This is probably bikesheddy but is there a reason `select:item': 'onSelectItem'` was not used instead. People tend to look for a standard naming convention for this stuff to avoid having to over think naming stuff.